RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.cc

```diff
@@ -0,0 +1,1038 @@
+    }     
+  } 
+
+  if (rechits.size() < 2) { 
+    return std::vector<CSCSegment>();  
+  } 
+
+  // We have at least 2 hits. We intend to try all possible pairs of hits to start  
+  // segment building. 'All possible' means each hit lies on different layers in the chamber. 
+  // after all same size segs are build we get rid of the overcrossed segments using the chi2 criteria 
+  // the hits from the segs that are left are marked as used and are not added to segs in future iterations 
+  // the hits from 3p segs are marked as used separately in order to try to assamble them in longer segments  
+  // in case there is a second pass    
+
+  // Choose first hit (as close to IP as possible) h1 and second hit 
+  // (as far from IP as possible) h2 To do this we iterate over hits 
+  // in the chamber by layer - pick two layers.  Then we 
+  // iterate over hits within each of these layers and pick h1 and h2 
+  // these.  If they are 'close enough' we build an empty 
+  // segment.  Then try adding hits to this segment. 
+
+  // Initialize flags that a given hit has been allocated to a segment 
+  BoolContainer used(rechits.size(), false); 
+  BoolContainer used3p(rechits.size(), false); 
+
+  // This is going to point to fits to hits, and its content will be used to create a CSCSegment
+  sfit_ = 0;
+
+  // Define buffer for segments we build  
+  std::vector<CSCSegment> segments; 
+
+  ChamberHitContainerCIt ib = rechits.begin(); 
+  ChamberHitContainerCIt ie = rechits.end(); 
+
+  // Possibly allow 3 passes, second widening scale factor for cuts, third for segments from displaced vertices 
+  windowScale = 1.; // scale factor for cuts 
+
+  bool search_disp = false;	   
+  int npass = (wideSeg > 1.)? 3 : 2; 
+
+  for (int ipass = 0; ipass < npass; ++ipass) { 
+    if(windowScale >1.){ 
+      iadd = 1; 
+      strip_iadd = 2; 
+      chi2D_iadd = 2; 
+    } 
+
+    int used_rh = 0;
+    for (ChamberHitContainerCIt i1 = ib; i1 != ie; ++i1) {
+      if(used[i1-ib])used_rh++;
+    }
+
+    //change the tresholds if it's time to look for displaced mu segments                                                                                                                                                          
+    if(doCollisions && search_disp && int(rechits.size()-used_rh)>2){//check if there are enough recHits left to build a segment from displaced vertices                                                                     
+      doCollisions = false;
+      windowScale = 1.; // scale factor for cuts   
+      dRMax = 2.0;
+      dPhiMax = 2*dPhiMax;
+      dRIntMax = 2*dRIntMax;
+      dPhiIntMax = 2*dPhiIntMax;
+      chi2Norm_2D_ = 5*chi2Norm_2D_;
+      chi2_str_ = 100;
+      chi2Max = 2*chi2Max;
+    }
+
+    for(unsigned int n_seg_min = 6u; n_seg_min > 2u + iadd; --n_seg_min){ 
+      BoolContainer common_used(rechits.size(),false); 
+      std::array<BoolContainer, 120> common_used_it = {}; 

```

Comment:
If I'm not mistaken 
``` common_used_it {} ``` will initialize to zeroes
```common_used_it = {}; ``` will leave elements of common_used_it uninitialized